tests/unit/lms/services/jwt_test.py

```diff
@@ -57,24 +61,90 @@ def test_encode_with_secret(self, svc, jwt):
         )
         assert encoded_jwt == jwt.encode.return_value
 
+    def test_decode_lti_token_with_no_kid(self, svc):
+        jwt_str = self.encode_jwt({"key": "value"}, headers={"key": "value"})
+
+        with pytest.raises(InvalidJWTError) as err_info:
+            svc.decode_lti_token(jwt_str)
+
+        assert "Missing 'kid' value in JWT header" in str(err_info.value)
```

Comment:
```suggestion
        with pytest.raises(InvalidJWTError):
            svc.decode_lti_token(jwt_str)
```

I'm a big fan of not checking strings in exceptions for a couple of reasons:

 * On principal the exception class should be enough to decide what to do (we don't / shouldn't catch and then inspect the string and then act). i.e. Something as fragile as a human readable string shouldn't be part of our interface, and if it's not part of our interface, we shouldn't be fixing it in place with a test
 * I don't like tests failing because you fix a typo or rephrase an exception
 * Code coverage should let us know if we've missed a case - If we hit them all "by mistake" somehow, and everything works... then let's take the win
 * I'm lazy! And it ends up with shorter tests
 
There is a plurality of opinion on this in the team though. I think my view is well summed up by an answer on SO:

https://softwareengineering.stackexchange.com/questions/384164/should-log-statements-be-tested

> No.
>
> Logging is a code aspect, not required behaviour: Your application would behaviourally work just as well without the logging.

No doubt a motivated thinker could come up with a counter-point.

Reply:
In an un-related note, if we went with the suggestion of collapsing all the behavior we want into the function, then this test would be:

```python
jwt_str = self.encode_jwt({"key": "value"}, headers={"key": "value"})

assert not svc.decode_lti_token(jwt_str)
```

Which would avoid the whole issue.